packages/catalog-model/src/kinds/ComponentEntityV1alpha1.ts

```diff
@@ -39,6 +39,21 @@ export interface ComponentEntityV1alpha1 extends Entity {
     consumesApis?: string[];
     dependsOn?: string[];
     system?: string;
+    /**
+     * A list of entity refs with kind=Environment, that this component is deployed to.
+     */
+    deploysToEnvironments?: string[];
+
+    /**
+     * A map of Environment entity ref to entity metadata to override in that environment. By default,
+     *  a Component's metadata reflects their production deployment.
+     */
+    environmentOverrides?: Record<
```

Comment:
I'm not 100% on this being a useful way to override information. It adds a massive amount of complexity for backend processors/providers/services but is useful for frontend switching of context like PagerDuty and Sentry. Open to ideas on improvements.